pkg/sshd/lock.go

```diff
@@ -26,11 +26,32 @@ type RepositoryLock interface {
 }
 
 func wrapInLock(lck RepositoryLock, repoName string, timeout time.Duration, fn func() error) error {
-	if err := lck.Lock(repoName, timeout); err != nil {
+	if err := lck.Lock(repoName); err != nil {
 		return errAlreadyLocked
 	}
-	defer lck.Unlock(repoName, timeout)
-	return fn()
+	tick := time.Tick(timeout)
```

Comment:
the ticker created here will tick forever, which doesn't seem to be what you want and will leak. it looks instead like you should create a [Timer](https://godoc.org/time#Timer)